src/utils/cachingAssets.ts

```diff
@@ -65,33 +68,51 @@ export const writeUrlInCache = ({ url, fileType, parentId, client }: WriteUrlInC
   })
 }
 
-export const writeVideoDataInCache = ({ data, thumbnailUrl, client }: WriteVideoDataCacheArg) => {
+export const writeVideoDataInCache = ({ edge, thumbnailUrl, client }: WriteVideoDataCacheArg) => {
   const video = client.cache.writeFragment({
-    id: `Video:${data.id}`,
+    id: `Video:${edge.node.id}`,
     fragment: VideoFieldsFragmentDoc,
     fragmentName: 'VideoFields',
     data: {
-      ...data,
+      ...edge.node,
       thumbnailPhotoUrls: thumbnailUrl ? [thumbnailUrl] : [],
       thumbnailPhotoAvailability: AssetAvailability.Accepted,
     },
   })
   client.cache.modify({
     fields: {
-      videos: (existingVideos = []) => {
-        return [video, ...existingVideos]
+      videosConnection: (existingVideos = {}) => {
+        return {
+          ...existingVideos,
+          totalCount: existingVideos.totalCount + 1,
+          edges: [{ ...edge, node: video }, ...(existingVideos?.edges ? existingVideos.edges : [])],
+        }
       },
     },
   })
 }
 
-export const removeVideoFromCache = (videoId: string, client: ApolloClient<NormalizedCacheObject>) => {
+type NormalizedVideoEdge = Omit<VideoEdge, 'node'> & {
+  node: Reference
+}
+
+export const removeVideoFromCache = (videoId: string, client: ApolloClient<object>) => {
   client.cache.modify({
     fields: {
       videos: (existingVideos = []) => {
         client.cache.evict({ id: `Video:${videoId}` })
         return existingVideos.filter((video: VideoFieldsFragment) => video.id !== videoId)
       },
+      videosConnection: (existing = {}, opts) => {
+        client.cache.evict({ id: `Video:${videoId}` })
+        return (
+          existing && {
+            ...existing,
+            totalCount: existing.totalCount - 1,
```

Comment:
This will result in decrementing `totalCount` regardless of whether the given video was actually included in that query cache, right? E.g. I delete public video but `totalCount` is decremented for queries targeting only unlisted videos as well. Correct?

Reply:
Yes, you're right - should be fixed now.